### ISSUE TITLE
Future of rails-assets

### ISSUE THREAD

**Issue by jandudulski** [tenex/rails-assets#291]
Hi all,

as you probably know rails-assets.org is currently hosted thanks to courtesy of ShellyCloud team but unfortunately they will shut down on March 31, 2016. There are at least two topics to discuss:
- where should we migrate?
- what should be the future of rails-assets?
## Migration

There is obvious deadline - March 31, 2016. Before that day we have to find hosting platform which would like to become a new home for rails-assets. We are going to contact with most known hosting platforms and ask for support but if you already know any which could help us - please leave a message.
## Future

rails-assets was born to fulfill some needs - we didn't want to manually build and maintain gems for vendor libraries with assets. However, world moved forward and there are a [lot](https://viget.com/extend/gulp-rails-asset-pipeline) of solid [solutions](https://medium.com/brigade-engineering/setting-up-webpack-with-rails-c62aea149679#.2qucp9x55) which allow to get rid of `rubygems` environment and separate backend and fronted libraries in rails projects.

The last nail in the coffin is [current state of bower](https://github.com/bower/bower/pull/1748#issuecomment-150104311) and generally speaking - [community movement from bower to npm based solutions](https://gofore.com/ohjelmistokehitys/stop-using-bower/). If you are not aware - rails-assets functionality base on bower packages.

Our current plan and proposal is to migrate rails-assets in its current form to the new platform and support it until end of 2016. After that we would like to migrate existing packages to kind of ftp so existing projects could still bundle but it won't work for packages which has not yet been transformed.

We are open for any comments, help offer and proposals.

**Comment by sheerun** [comment]
Bower is not dying: http://bower.io/blog/

**Comment by sheerun** [comment]
I've already proposed a solution, but we'd need a sponsor or a crowdfunding campaign for it (Monterail has no resources): Create a gem that works like rails-assets, but allows to fetch files directly from GitHub.

**Comment by teamon** [comment]
@sheerun I am happy to continue this topic in separate thread

**Comment by teamon** [comment]
As discussed before, we don't really see a bright future for rails-assets as the perfect solution to assets management in rails apps. As @jandudulski said, there are now much better tools for that.

Monterail will continue to support the current state of rails-assets including migration to new hosting.

Having said that, despite rails-assets was started by me and others (@sheerun, @porada) who were then part of Monterail's team, neither the company nor I claim to be the ultimate decision makers here. rails-assets is still an open source project and we are open to ideas and contributions from the community.

To put things straight:
- we (Monterail) want to support current functionality until the end of 2016
- we want to support some kind of read-only mode since the end of 2016
- we do not believe creating serverless rails-assets gem is the good way to go
- instead, we want to provide as many resources as possible for everyone to migrate to other tools

**Comment by ch1c0t** [comment]
> we do not believe creating serverless rails-assets gem is the good way to go

Could you please elaborate more on that? Why is it a bad idea?

To me, it is so convenient to use only one tool(bundler) to manage all dependencies. I probably would like to have a rake task gem for converting an asset to a gem.

**Comment by teamon** [comment]
If you take a look at https://github.com/rails-assets/rails-assets/tree/master/app/models/build you can see it is quite complex processing.
Then, here https://rails-assets.org/status you can see the various error that are not yet handler properly.
This is all due to fact that bower packages are extremely inconsistent.

The front-end stack has changed a lot in the last few years, we now finally have standard access to proper modules and code isolation in the browsers. In the same time, sprockets is (and probably will) be a _simple_ tool for joining files together. 

Tools like sprockets or rails-assets will always be a bit behind the JS world.

In our opinion rails-assets did it's job over the last 2,5 years, but it won't be needed soon since tools like npm/webpack are already doing a great job and there is no need for additional layers.

**Comment by ldonnet** [comment]
Hi,

I used rails-assets solution for 1 year. I'm agree with you it's not the perfect solution. Rails assets deals with javascript library and it's not an easy task. For me it's also the best solution I have found to get :  
- packages up to date
- one package installer bundler
- not ask 30 owner of repositories to make an update of javascript libraries in the gem. 

Perhaps the future is npm/webpack tools but they introduce also a new complexity in rails application.

**Comment by drewhamlett** [comment]
If you guys think the Bower landscape is somehow bad try to install modules from NPM.  Most of the time the module isn't even available on NPM.  Not being on NPM means it doesn't use UMD, and you have to shim your shim so you can shim while you shim.  It's a terrible place out their for front end devs. I was doing it full time for a year.  Half the time you just do

window.$ = require('jquery')
require('jquery-ui/wtf-am-i-even-doing') 

Or just copying javascript from a repo and plopping it in your project.

Gulp/Browserify/Webpack is cool for those just getting into programming and want to do everything manual.  They want to sit around and engineer solutions just to engineer them.  That's cool when you're 10 but when you grow up and get a real job you may have to do things.  While most people are sitting around comparing their Gulp buildscript sizes, I'm trying to build stuff.

Rails assets solved a huge problem.  We had those stupid ass wrapper gems, and it replaced them.  It's a shame that the over engineered world of the front end has leaked over into this project.

I don't want to roll my own asset pipeline with Gulp because I like my free time.  

I do think there is room for a browserify-rails which is a decent replacement for rails assets, since you use NPM.  It's slow as molasses, but hey whatever can make a front end devs life harder, they'll be up for it.

All joking aside, I think work should go into Browserify rails to make it faster and detect file changes on save instead of browser reload.  I don't think the future is to replace the entire asset pipeline.  That means image/font management.  Just replace the javascript piece.

**Comment by drewhamlett** [comment]
Just to show how over engineering has reached new heights on the front end.  Take a look at this.

https://www.airbnb.com/rooms/259576

Using react for static content on that page.  Provides no interaction to user.  It's just static.

This brand asset page for Uber.

http://brand.uber.com/

Probably have 1000 engineers twiddling their thumbs, so they create the most over engineered brand asset page of all time.

**Comment by ajb** [comment]
I don't have much to add in terms of substantive discussion here, but I wanted to drop in on the conversation in order to give some massive gratitude and :heart: :heart: :heart: to everyone who's worked  on rails-assets. 

I know that the world of frontend development moves fast, and some of you might feel that rails-assets has been surpassed by npm, webpack, gulp, or whatever the latest and greatest is. 

However, I'll just add that our team is _extremely happy_ with rails-assets, and assuming that it stays online (or that we can stand up our own server), we'll be planning to use it well into 2016 and beyond.

**Comment by halilim** [comment]
I sometimes call myself "full-stack" but I'd consider myself as a 0.75-stack (0.5 backend, 0.25 frontend). Or, the projects I'm working on are not frontend-heavy. Hence I have less time to apply every new frontend tech I've heard of to the project at hand. Instead I find peace in moderate solutions like Rails Assets which, a) doesn't have the disadvantage of wrapper gems, and b) doesn't have the complexity of full-blown solutions like Gulp, Webpack etc.

So until the time new frontend mechanisms get really widespread, Rails Assets fills an important gap for me and I'd appreciate it if it continues to do so for the foreseeable future.

**Comment by Gedrovits** [comment]
Totally agree with @halilim . Yeah, JS world moving fast, but for simple apps, which depend on simple, non-fancy-js-includes-requires libraries, this is a better way then old 'asset gems' like 'blablabla-rails'.

Maybe it's not ideal solution, but it's definitely filled the gap for us. So it's sad it may be closed.

**Comment by colbywhite** [comment]
@teamon, @sheerun, and/or @jandudulski, was there any resolution to this? Based on the lack of activity on the repo and this issue, it appears that this is on pace to fade away in March? Is that an accurate assessment?

**Comment by teamon** [comment]
@colbywhite Currently we are arranging a new hosting. As previously stated we will support the current state at least until the end of this year.

**Comment by drewhamlett** [comment]
I'm not sure a lot of people know it's going to shut down.  Seems to be quite a few people using it.  I really love this service.

**Comment by ldonnet** [comment]
As @drewhamlett said I think this news could be more spread. A ticket in github is very specific, why not communicate on the front end services to get more reactions from end user : https://rails-assets.org/.

**Comment by boone** [comment]
It might help to post something on the home page. Someone might be able to step in and offer hosting. What are the hosting needs?

I'm also a happy user of Rails assets. I haven't found the other solutions compelling enough to want to switch.

**Comment by hayesgm** [comment]
I would say that rails-assets fills an important niche, esp. in light of `react-rails` deciding to stay sprockets instead of using webpack, browserifying, etc. For better or worse, so long as rails stays on sprockets as default, gem based assets are going to be a thing long into the future.

What help can the community give?

**Comment by drewhamlett** [comment]
I've been giving quit a bit of thought to the front end ecosystem lately.  I think people are frustrated with the NPM ecosystem leaking into the Rails.  It's pretty much the anti thesis of the Rails way.  A lot of the .NET guys are feeling the same too.  A new Web MVC project brings in NPM for all assets.

I think NPM is the future of front end but I think we can work with it.  There are a lot of good projects coming out.  React, Webpack, Babel, but the Rails community has made the configuration hell easier. For example react-rails.  You can get Babel and ismorphic React setup in a few lines of code.  

If we could manage NPM assets with the Gemfile as you do with rails-assets(bower), we would get locked dependencies by default.  Another instance is changing sprockets to support require(or import).  There's a lot of fragmentation going on in the JS world because developers just start a new projects instead of fixing the older stuff.  I think in the Rails community we have a decent track record on rallying behind one project.

**Comment by colbywhite** [comment]
@drewhamlett, are you suggesting that the rails community should go away from bower and focus on integrating with npm? As in making an equivalent of this project, but one that focuses on npm?

**Comment by bbraga** [comment]
https://rails-assets.org is down! Houston we have a problem

![screenshot-rails-assets org 2016-01-17 10-11-25](https://cloud.githubusercontent.com/assets/185834/12376702/b657bf0e-bd02-11e5-90e7-7541ca378f6e.png)

PS: Due to this I believe other external mirrors for assets would be a very nice to have, in case of catastrophic failure such as today

.

**Comment by bsbodden** [comment]
I for one hate dealing with NPM and found bower a much better route. Then finding rails assets made it seamless. Why rather than having a full-blown gem source/server we just keep a list of converted bower packages and push/publish them directly to rubygems.org?

**Comment by pauloschilling** [comment]
And rails-assets.org is down again right now, same as yesterday, "504 Gateway Timeout" from Shelly Cloud.

I truly believe you need to bring the future of rails-assets.org to the table, but what about the present? In the meantime, there's nothing you can do to make the current service more stable and reliable? I know you depend(ed) on Shelly Cloud's courtesy to host everything, but even if they had no plans to shutdown the service you should be considering doing something about it.

Talking about the tool itself, we already removed rails-assets from our most critical project, and now we are going to do the same to the last one. When you have dedicated front-end developers you'll want to avoid such tools, and start using gulp, bower, npm, grunt and so on, like @jandudulski said, and you'll end up with separated frontend and backend dependencies, which is really good.

Please, I really don't want to be rude, I just want to contribute with our thoughts and the internal decisions we have made regarding this.

Oh, and have you ever considered talking to someone behind bigger rails-related projects? Maybe one of them can help you either by offering the resources you need to keep the service online, or even by adopting it to integrate with their main service.

Also, I want to thank you for all the effort you had to make and keep rails-assets running, we really appreciate that!

**Comment by bsbodden** [comment]
On the meantime folks can use https://github.com/rharriso/bower-rails/, not as convenient but it'll do.

**Comment by drewhamlett** [comment]
@pauloschilling Using Grunt and Bower right now would be akin to Luke Skywalker Tauntauning a dead corpse to live in.  I don't really mind Bower per say, but it seems NPM has won(for better or worse).  I personally don't want dedicated front end people working for me.  I hire full stack developers who are very good at Javascript.  This tends to keep the app balanced, without the over engineering front end parasites(hint: I was one) trying to tackle every problem ever with a gigantic helping of shitty Javascript and flavor of the month frameworks.

Anyway, what I'm getting at, is not everyone has the resources to have dedicated front end engineers.  People are fine(and like) using the asset pipeline and Bundler.  We just need a good solution.  I don't mind using NPM to manage dependencies but I don't want to manage my own asset pipeline.  Sure I'd rather use Bundler, but it seems the front end world has leaked everywhere at this point.  I really like the project https://github.com/browserify-rails/browserify-rails, it's just too slow for me at this point.  Maybe the solution is to iterate on that and make it better.

@colbywhite : I think that's a possibility.  It depends on if there's any interest in it.  It can't just integrate NPM though.  It would need to be an out of the box solution for managing NPM dependencies and wrapping Webpack or Browserify.

**Comment by mattbrictson** [comment]
> I don't mind using NPM to manage dependencies but I don't want to manage my own asset pipeline.

@drewhamlett I agree 100%. Have you tried a little project called [Torba](https://github.com/torba-rb/torba)? Its creator describes it as "Bundler for Sprockets". Uses NPM rather than Bower.

I just ported a small project from Rails Assets to Torba and it worked on the first try. Definitely promising!

**Comment by pauloschilling** [comment]
@drewhamlett, I agree with you, but different projects have different needs, and different countries have different people. Here in Brazil a full stack engineer normally don't have neither the knowledge nor the passion to maintain the frontend (read CSS, HTML, part of the JS), so we **have** to hire people "just" to do that in order to have a good UI/UX.

This is just a parentheses on why we ended up with dedicated frontend engineers, and at least in our case it was not a question of having resources (money) for it, but the real need to do that because people have different abilities here.

Anyway, I don't want to change the main topic, sorry about that, but I think this is also good to better understand what kind of projects and teams can take fully advantage from solutions like rails-assets, and the "full stack" thing probably is the key.

The "X is better or worse than Y" will always exist, and this is good as well, the key again is what is better for you and for projects like yours.

Is pretty clear that rails-assets is most recommended for full stack teams, and I hope they can find the right path to keep it running, no matter if there's a "better" tool or not.

Best,

**Comment by ajb** [comment]
A couple of thoughts:
- To the maintainers: @dobtco is willing to offer some level of financial sponsorship for hosting. Feel free to contact me via email to chat more about this. (You can find my address in my git logs or on my personal website.)
- To everyone else: hosting another instance of rails-assets shouldn't really be that hard. Maybe someone in the community can step up to offer a Dockerized version of the service that would be easy to deploy and maintain? I recently stood up an instance on my own, and while in some ways it's just a basic Rails app, there do seem to be a few tricks to getting everything running smoothly. Docker would make this part trivial.

**Comment by sheerun** [comment]
I can setup hosted version of Rails Assets on AWS and support it, if someone is interested.

I need to see it can profitable, so please fill this form: https://docs.google.com/a/sher.pl/forms/d/16EQ1FdInLOuEZ_E9Rt0pFy54QnLzKSw7JCVbDYewCTw/edit?usp=forms_home

**Comment by hut8** [comment]
@sheerun Google says that form is restricted.

I spent almost an entire night during this outage looking for a better alternative, and my findings were basically that the whole ecosystem of front end assets is somewhat of a mess. I think that this project is actually a really good solution even if it's not perfect. I would be happy to host it as long as I can afford it out of pocket. Can someone familiar with the current hosting please describe the resources used in terms of disk space, bandwidth, memory, etc? I want to price it out and maybe find some areas for optimization. I'm sure I can take this on by mid 2016.

**Comment by sheerun** [comment]
Currently it's 30 GB SSD, 600 GB traffic/month, but growing. I've updated the form link:
https://docs.google.com/a/sher.pl/forms/d/16EQ1FdInLOuEZ_E9Rt0pFy54QnLzKSw7JCVbDYewCTw/edit?usp=forms_home

**Comment by rjhancock** [comment]
I've sent a message off to DigitalOcean as they work with non-profits and OS projects for either discounted/free hosting. Their $40/mo server should be able to handle it (60G SSD / 3T Bandwidth). I'd offer to pay for it myself but don't have the spare $40/mo right this second.

**Comment by hut8** [comment]
I just sent an email to @jandudulski offering to have my consultancy host (and more importantly support) it for free. I can definitely have this set up by next week.

**Comment by sheerun** [comment]
Hosting isn't most expensive part. Support is.

**Comment by rjhancock** [comment]
Support wise, not much I can do. But if the hosting is taken care of thats one less thing to worry about.

**Comment by sheerun** [comment]
@hut8 We can't pass hosting that easily, for security reasons. It would give your consultancy a power to change any package's contents. You can setup separate service though, if you want.

**Comment by joshjordan** [comment]
@sheerun I think @hut8 is offering hosting _and_ support.

Can you explain the difference in security at the current host and the one being proposed?

**Comment by sheerun** [comment]
It's the same reason npm won't give some random agency a write access to all of their registry.

// or github to all of their public repositories

**Comment by joshjordan** [comment]
What is that reason, and how is it different for the current ownership?

**Comment by bsbodden** [comment]
Has anybody approach the Rubygems team directly? I wonder what their take on hosting asset gems is? I mean, nothing prevents somebody from using the Gem generation portion of the rails-assets app and simply publish the generated gems to rubygems. It would be nice if the bower search/request process was backed into RubyGems but of course that opens the door for them to have to do bower, npm, etc.

**Comment by themgt** [comment]
The way Phoenix (Elixir) uses Brunch.io seems much preferable, but unless/until someone ports similar functionality into Rails, rails-assets.org is will remain hugely useful.

I'd encourage a move to rails-assets.org being community-run and using easily deployable site mirrors - we at @pogodan would be happy to run one.

**Comment by stympy** [comment]
We at @honeybadger-io would be happy to help with hosting as well

**Comment by yannski** [comment]
At Scalingo https://scalingo.com/ we can help too.

**Comment by teamon** [comment]
Hi everyone and thanks for all your comments. I won't be able to answer all of them, but let me tackle at least few points here:
1. We do realize that rails-assets is useful for quite some people (including Monterail itself). We still use it in some of our projects and we do want to support it in the current form **at least** until the end of this year.
2. As already mentioned there are two topics here: the future (2017) and the present, but both are very tightly connected. Right now we are focusing on arranging new hosting. I'm extremely happy that so many of you offered their support with that - please let me get my head around it and I'll contact you each individually.
3. In terms of ownership - currently few people have access to production machines and that includes Shelly Cloud team, Adam, Jan, me and few other Monterail developers. In theory, we _could_ mess with the packages, but this would be a rather suicidal path for the individuals and Monterail in the ruby on rails community. And it would be the same case with any other company that supports/hosts the project.
4. Before rails-assets was born, we wanted to automate the process of creating assets gems and push them to rubygems.org. One of the reasons we stopped that was to avoid pollution rubygems with all those gems. It could probably solve some of the performance issues, but I think we still can do better than this.
5. We definitely should let people know about the plan, preferably by some announcement on the home page. ~~I'll try to get this done soon. (and I'm definitely willing to accept a PR if someone does this before me :) )~~ https://github.com/rails-assets/rails-assets/pull/308
6. Finally, I do understand your disappointment with the recent quality of the service and the plans for future. I hope we can sort this out together as a community and prove one of the most important foundations of open source software - one can pick up when the others have left.

**Comment by hut8** [comment]
I saw that the existing deployment-related stuff is (understandably) tightly coupled to ShellyCloud. I wrote Ansible and Capistrano scripts last night and have actually deployed this application on a subdomain of my own. There are a couple minor issues I haven't fixed yet (externally hosted fonts giving 403s, lack of installation in playbook for Sidekiq Upstart scripts, etc). My deployment also includes Let's Encrypt, which I found was really neat.

I'm also going to write a quick script to try to import the existing converted packages from https://rails-assets.org/components.json, mostly as a test.

**Comment by andrewsomething** [comment]
Hey all! Looks like you have a lot of good options for hosting; add DigitalOcean to the list of folks willing to pitch in. Reach out to us opensource@digitalocean.com if we can help.

**Comment by nhance** [comment]
Hey, I may be willing to sponsor this for @railsasaservice. Can you give me a sense of the monthly costs involved?

**Comment by hut8** [comment]
https://rails-assets.tenex.tech is up now. I'm loading all the components present in https://rails-assets.org/components.json so expect slowness, but I'm eager to hear some feedback.

**Comment by lubekpl** [comment]
@hut8 Can you provide instructions for deployment?

**Comment by justintanner** [comment]
First a huge thank you to everybody who worked on and is currently maintaining rails-assets. You have saved me precious development time.

Please consider continuing support for rails-assets beyond 2016. For rails apps that have a handful of javascript deps rails-assets is an elegant solution. The alternatives are very complex and over engineered.

Please reconsider abandoning rails-assets.

**Comment by jvanbaarsen** [comment]
@jandudulski What are your main concerns at the moment? The lack of contributors? Or the costs of the hosting? For the latter I have seen a lot of suggestions in this topic already (Digitalocean wants to chip in, so is Honeybadger).

If it is about contributors, what are the current pain points that need to be addressed? I'm sure we can come up with a new fresh set of contributors. If needed, I can free some of my time to help out in the transition period.

Would you be open to work with me and others to see if we can keep rails-assets alive and going? I'm sure we can find the community support it needs, without you having to sacrifice all of your freetime.

**Comment by guizmaii** [comment]
Hi @hut8,

Great initiative to host a copy but I can't proxify it in my Nexus because (I think) of your SSL configuration that is not so strong (see SslLabs results : https://www.ssllabs.com/ssltest/analyze.html?d=rails-assets.tenex.tech&hideResults=on).
Can you improve it maybe ? Mozilla gives us a tool to help us config this well : https://mozilla.github.io/server-side-tls/ssl-config-generator/

I think that just replace your DH key for a 2048 bits one will improve your grade well.

**Comment by guizmaii** [comment]
@hut8 here some feedback : my build is blocked since 10 minutes to this step : `Fetching version metadata from https://rails-assets.tenex.tech/`

This is my use of your repo : 

```
source 'https://rails-assets.tenex.tech' do
  gem 'rails-assets-oncapslock'
end
```

:'(

**Comment by richpeck** [comment]
We've used `rails-assets` for over 1 year, it's the best solution of its kind.

What are you looking for to make it work? Money, bandwidth, server space, or all 3?

**Comment by jandudulski** [comment]
@jvanbaarsen 

> @jandudulski What are your main concerns at the moment?

Our main concern is that we're going to withdraw from using rails-assets and it is hard to maintain project which you're not using anymore. But we don't want to cut the project and leave developers unhappy with the situation so we're working on to find a new maintainer and I believe we would be able to announce some good news soon.

**Comment by jvanbaarsen** [comment]
@jandudulski Ok great! If needed, I can step up as well. Jus let me know :-)

**Comment by hut8** [comment]
@guizmaii thank you so much for your feedback. I'm opening an issue on my clone of the repo and I'll have it fixed by the end of the day. I know that it's really slow at the moment; this is because I'm trying to build all the packages currently present on rails-assets.org. This is probably a bad idea, so I'm stopping it now.

**Comment by bhgames** [comment]
Hey guys, I just spent about 4 hours configuring Torba to work in place of Rails Assets. It's a pain in the ass to get setup because you have to find the right js file in each pulled repo to include instead of the nicer rails assets way, but it doesn't rely on any third party hosted service. 

I think it's a good "drop-in" replacement, especially in that it carries NPM support, and all you need to do to add a new NPM module is drop it in, and then inspect your ~/.torba/new-npm-module-647433/ directory for the proper JS file to include. Normally this is in the dist folder but not all npm modules have them. It's kind of a mess over there in JS land.

Edit: Link to Torba https://github.com/torba-rb/torba

It also as of now has good Heroku documentation and support, though I haven't actually tried productionizing it yet. That's coming up next. Will report back if successful.

**Comment by colbywhite** [comment]
> Our main concern is that we're going to withdraw from using rails-assets and it is hard to maintain project which you're not using anymore.
> ~ @jandudulski 

If you don't mind me asking, if you're not using rails-assets, what are you using? A more direct back-end agnostic tool like gulp/browserify/etc.?

**Comment by rbrigby** [comment]
We at retail.gamegear.com use rails-assets in a new app we are building.  We really do not want to migrate away from it if possible.  It is by far the best solution for us.

**Comment by teamon** [comment]
Small update: we are setting up a mirror with @hut8, stay tuned :)

**Comment by brooks** [comment]
Can anyone confirm what date rails assets will be going away? The first post here says March 2016 but Im seeing end of 2016 on twitter, etc. Thanks :)

**Comment by hut8** [comment]
Hi @brooks ! rails-assets is not going away. I know the thread is long, so I'll give you a TL;DR version.

The company that's hosting rails-assets (shellycloud) is shutting down in March. Monterail, the company that developed `rails-assets`, will make sure it is up till the end of 2016.

My consultancy, [Tenex](https://www.tenex.tech) uses this and has forked it to https://github.com/tenex/rails-assets . As @teamon said 6 days ago, we have set up a mirror at https://rails-assets.tenex.tech . @joshjordan and I (of Tenex) have no plans of abandoning this.

So in summary, it will not be going away :smile:

**Comment by brooks** [comment]
Thanks @hut8  :smile:

**Comment by mvastola** [comment]
I'm sorry to hear `rails-assets` is planning to close its doors. Unfortunately, I can't comment much on what is being debated about the declining need for rails-assets (ostensibly due to the existence of better alternatives) as I don't really have experience with alternatives.. I -- as did, I imagine, many current users of this service -- went straight from manual asset management to `rails-assets`. 

As such, could anyone be so kind as to lay out some of the many alternatives to `rails-assets` (alluded to by @jandudulski in his OP) providing both dependency management and `rails` asset pipeline integration while also highlighting any notable features or differences? (Perhaps such a summary would even be worth linking to from [rails-assets.org](https://rails-assets.org/)?)

**Comment by jvanbaarsen** [comment]
@hut8 Will you be taking over the domain as well?

**Comment by hut8** [comment]
Good question @jvanbaarsen . I've discussed this with @teamon and he said that if all goes well with our hosting of it, then yes, but it would be cool if he could confirm that :smiley: 

Also, @mvastola , I wouldn't say it's shutting down for the reasons I gave a few hours before you made your post. It's simply that a different company is maintaining it. We've already made a few minor improvements, and have deployed it on DigitalOcean, who is sponsoring it as well.

I too would be interested in hearing the alternatives. After all, I would hate to waste my time contributing to something worthless :stuck_out_tongue:  When `rails-assets` was down for a weekend, I did an enormous amount of research about this, and decided in the end of that `rails-assets` is definitely the best solution for a rails project who wants the frontend assets to "just work". I switched to `bower-rails` for a bit, and unfortunately still have it in one project, but it's caused me a ridiculous amount of maintenance headaches. For example, not only does it introduce more server-side dependencies during deployment, but it doesn't lock in versions, so I had deployed Angular 1.5 when `angular-ui-select` only works with <= 1.4.9 without a kludge (https://github.com/angular-ui/ui-select/issues/1325). So I can speak about the alternatives too, but can't say much without using expletives.

Probably a better solution for projects in general would be completely de-coupling the frontend from the backend, to the point where they could easily be in different git repositories. But that's definitely overkill for most use cases I've had. **The lack of good alternatives for simple use cases is why I chose to pursue hosting and maintaining this excellent tool**

**Comment by shadowbq** [comment]
Not worthless. Can we use this new mirror (https://rails-assets.tenex.tech) ? Can you post something about the domain itself, whether it will be temporary or a production mirror. Does bundler follow redirects if you take over the existing domain? Will you maintain this mirror on this domain for at least six months? [So in summary, it will not be going away] - The project/the domain or both? Thanks for stepping up.

**Comment by marianoviola** [comment]
@hut8 thank you for your commitment. May I suggest to add a donation button somewhere in the new website? I would be very happy to contribute with some money to the project.

Edit:
Are you thinking to a monetisation strategy? I mean something to support private packages or some plans based on usage (bandwidth, number of packages and so on).

**Comment by mensfeld** [comment]
@hut8  please add a donation button. Rails assets are great for prototyping and testing! I will gladly donate this!

**Comment by hut8** [comment]
@shadowbq Please do use the mirror! We will be supporting it for the foreseeable future. `rails-assets.org` does not currently point there, although at some point @teamon had mentioned to me via email that Monterail would be willing to transfer the domain if everything is stable with our mirror. It's on ShellyCloud till March, so I'm thinking that might be a good time to switch. We will continue using the https://rails-assets.tenex.tech domain indefinitely. If you find issues specific to our mirror, please file them in our fork (https://github.com/tenex/rails-assets)

@marianoviola @mensfeld Good suggestion! 

Edit: @shadowbq -> https://github.com/bundler/bundler/blob/master/lib/bundler/fetcher/downloader.rb#L22 yes, redirects are followed :+1:

**Comment by hut8** [comment]
@marianoviola @mensfeld Last night we added a donation button. See the top bar of https://rails-assets.tenex.tech

Again, thanks for the suggestion :+1:

**Comment by mensfeld** [comment]
@hut8 "You did not set a valid publishable key. Call Stripe.setPublishableKey() with your publishable key. For more info, see https://stripe.com/docs/stripe.js" :D

**Comment by marianoviola** [comment]
@hut8 great! You need to set a publishable key as suggested by @mensfeld :smile:

**Comment by hut8** [comment]
Hey! This was actually a test for our MVP of the donation system. Now that we know that people actually will click on the button, we have implemented it :wink:  It should be launched now. :rocket:

**Comment by mensfeld** [comment]
@hut8  works - you can check if the payment has been processed frm me ;)

**Comment by lxsameer** [comment]
`rails-assets` is one of my most important tools, I surely will miss it. Personally i don't like get my hands dirty with tools like `bower` or `npm` ( which is the worst piece of software i ever seen ).  Managing front end dependencies using `bundler` is a brilliant idea. I think rails-assets can continue its work and I love to see this happen.

**Comment by hut8** [comment]
Hey there @lxsameer . You don't need to miss it! I'm not sure if you read the entire thread since it's so long but `rails-assets` is definitely not going anywhere. It's sponsored by DigitalOcean, and myself and @joshjordan plan on supporting it indefinitely, certainly past the end of 2016. Feedback for our mirror at https://rails-assets.tenex.tech would be greatly appreciated.

**Comment by lxsameer** [comment]
@hut8 awesome. I'm very happy right now, Kudos to you guys.

**Comment by colbywhite** [comment]
Should this issue be closed with a recap comment? It appears that the new mirror will be the answer.

**Comment by mensfeld** [comment]
@colbywhite I believe this should be closed with a proper annotation about support option.

**Comment by hut8** [comment]
@teamon Can you comment about the domain? Are you satisfied with Tenex taking things from here?

**Comment by teamon** [comment]
Sorry for the radio silence!

About the domain - yes, definitely. Plus I think we can close this thread now.

I've created another issue #312 to track what's left to be done before March 31.

**Comment by assembler** [comment]
You should probably add message on rails-assets.org that people should use rails-assets.tenex.tech instead and not having to read this huge thread :)

**Comment by shadowbq** [comment]
Please edit the issue to include updated information on the top of the big thread

**Comment by MtnBiker** [comment]
Another shout out for rails-assets. This worked today for this newbie. I hadn't been able to push to Heroku for a few weeks and decided I had to get it done. Read discussions about asset managers. Tried some things and decided to give rails-assets a go even though I may to make adjustments soon. It worked for me. Thanks and keep up the good work.

**Comment by dt1973** [comment]
Used to have high hopes for https://github.com/half-pipe/half-pipe but I don't know what happened.

cc: @joefiorini

**Comment by joefiorini** [comment]
@dt1973 Thanks for the mention. I started the project but left the company where we were using it and went to a Python shop that obviously didn't have much need for it. I've also stopped using Grunt in favor of Webpack and other tools made for being a build solution. I'm happy to hand it off to someone if anyone wants to dust off the cobwebs and bring it into the modern age.

**Comment by klippx** [comment]
Webpack is the future boys

**Comment by mockdeep** [comment]
You know the future?!

**Comment by halilim** [comment]
The problem is, we haven't reached that future yet and we need a hassle-free asset management solution right now which doesn't change every six months :)

**Comment by tibastral** [comment]
And when you are back from the future you enjoy using rails-assets which is awesome :)

**Comment by milushov** [comment]
https://rails-assets.org/ is down? :(
http://www.downforeveryoneorjustme.com/rails-assets.org
![image](https://cloud.githubusercontent.com/assets/424326/15140694/001d4932-16a5-11e6-9f7b-ec9d8011d46a.png)

**Comment by richpeck** [comment]
Down for me too

Webpack looks good, similar to a lot of others like babel etc. Will be
interesting to see if a single package manager is adopted

**Comment by kofronpi** [comment]
https://rails-assets.tenex.tech/ is down as well. Maybe they're doing the transfer they talked about ?

**Comment by ssaunier** [comment]
Or Maybe `rails-assets.tenex.tech` is suddenly being hit with a lots of requests!

If you have urgent deployments to do, you can vendor the gems:

``` bash
bundle package
```

That way you'll have a `vendor/cache` folder created with every gem. You can commit / push that. Later on think of removing this commit (not just `revert`) as it grows your git repository a lot.

**Comment by kofronpi** [comment]
@ssaunier both URI point to the same server, see https://github.com/rails-assets/rails-assets/issues/329#issuecomment-218106608

**Comment by sjauld** [comment]
rails-assets.org is BACK!

**Comment by joshjordan** [comment]
@milushov @richpeck @kofronpi @ssaunier @sjauld thanks for the heads up. No transfer in progress - rails-assets.org is here to stay and shouldn't require any extra effort on your part. We weren't taking any more traffic than usual last night, and syslog isn't showing anything useful. Looks like we went hard down at 3:41AM. I'm following up with our hosting provider to get more info. Right now, we appear to be up and stable.
